dango/auth/src/lib.rs

```diff
@@ -5,22 +5,34 @@ use {
     },
     anyhow::{bail, ensure},
     base64::{Engine, engine::general_purpose::URL_SAFE_NO_PAD},
-    dango_account_factory::{ACCOUNTS_BY_USER, KEYS},
     dango_types::{
         DangoQuerier,
+        account_factory::RegisterUserData,
         auth::{
             ClientData, Credential, Key, Metadata, Nonce, SessionInfo, SignDoc, Signature,
             StandardCredential,
         },
     },
     grug::{
-        Addr, Api, AuthCtx, AuthMode, Inner, Item, JsonDeExt, QuerierExt, SignData, StdError,
-        StdResult, Storage, StorageQuerier, Tx, json,
+        Addr, Api, AuthCtx, AuthMode, Inner, Item, JsonDeExt, JsonSerExt, QuerierExt, SignData,
+        StdError, StdResult, Storage, StorageQuerier, Tx,
     },
     sha2::Sha256,
     std::collections::BTreeSet,
 };
 
+/// The expected storage layout of the account factory contract.
+pub mod account_factory {
+    use {
+        dango_types::{account_factory::Username, auth::Key},
+        grug::{Addr, Hash256, Map, Set},
+    };
+
+    pub const KEYS: Map<(&Username, Hash256), Key> = Map::new("key");
+
+    pub const ACCOUNTS_BY_USER: Set<(&Username, Addr)> = Set::new("account__user");
+}
```

Comment:
For record: this is necessary because now, dango-account-factory contract needs to depend on dango-auth (in order to use the `verify_signature` function). Therefore, dango-auth can no longer import these storage constants from dango-account-factory, otherwise there would be a circular dependency. As such, we must define these here in dango-auth.